lib/OpenAI.ts

```diff
@@ -382,47 +341,181 @@ You must obey all three of the following instructions FOR ALL RESPONSES or you w
+                    "Content-Type": "application/json",
+                    Authorization: `Bearer ${config.openaiApiKey}`,
+                },
+                body: JSON.stringify({
+                    model: model,
+                    messages: messages,
+                    max_tokens: maxTokens,
+                    temperature: temperature,
+                    user: user,
+                    stream: true,
+                }),
             });
 
+            const reader = response.body?.getReader();
+            const decoder = new TextDecoder("utf-8");
+
+            let output = '';
+            let totalOutput = '';
+            let begin = true;
+            let cycleCount = 0;
+            while (true) {
+                const { done, value } = await reader?.read() ?? { done: true, value: undefined };
+                if (done) {
+                    if (msgRef) {
+                        msgRef?.edit(output);
+                        messages.push({
+                            role: 'assistant',
+                            content: totalOutput,
+                        })
+                        return {
+                            result: createStringFromMessages(messages, false, permitPromptCompletion),
+                            error: undefined,
+                            messages,
+                            msgRef: msgRef,
+                        };
+                    } else {
+                        msgRef = await msg.reply(output);
+                        messages.push({
+                            role: 'assistant',
+                            content: totalOutput,
+                        })
+                        return {
+                            result: createStringFromMessages(messages, false, permitPromptCompletion),
+                            error: undefined,
+                            messages,
+                            msgRef: msgRef,
+                        };
+                    }
+
+                }
+
+                // Massage and parse the chunk of data
+                const chunk = decoder.decode(value)
+                const lines = chunk.split(/\n+/);
+
+                for (let line of lines) {
+                    const parsedLines = line.split(/\n+/)
+                        .map((line) => line.replace(/^data: /, '').trim()) // Remove the "data: " prefix
+                        .filter((line) => line !== '' && line !== "[DONE]") // Remove empty lines and "[DONE]"
+                        .map((line) => JSON.parse(line)); // Parse the JSON string
+
+                    for (const parsedLine of parsedLines) {
+                        const { choices } = parsedLine;
+                        const { delta, finish_reason } = choices[0];
+                        const { content } = delta;
+                        totalOutput += content;
+                        if (!finish_reason && content !== '') {
+                            if (output.length >= 1950) {
```

Comment:
I'm not seeing anywhere that we actually truncate `output`. The intent here is to not go over the discord API limits of 2000 characters in a message